client/src/pages/metadata-query/data-source-query.tsx

```diff
@@ -0,0 +1,106 @@
+import { useGetDatasources } from '@/api/datasource/hooks/use-get-datasources';
+import React, { useEffect, useState } from 'react';
+
+export const SourceQuery = ({ validator, queryName, handleQueryValid, handleQueryInvalid }) => {
+  const { apiQuery, blobQuery } = useGetDatasources();
+  const [selections, setSelections] = useState({});
+
+  useEffect(() => {
+    if (apiQuery.data || blobQuery.data) {
+      setDataSources();
+    }
+  }, []);
+
+  useEffect(() => {
+    let checkedSelections = Object.keys(selections).filter((item) => selections[item]?.selected);
+    handleSelection(checkedSelections);
+  }, [selections]);
+
+  const setDataSources = () => {
+    const dataSources = {};
+
+    blobQuery.data?.forEach((item) => {
+      dataSources[item.name] = { name: item.name, account: item.account, container: item.container, selected: false };
+    });
+
+    apiQuery.data?.forEach((item) => {
+      dataSources[item.name] = { name: item.name, account: item.account, container: item.container, selected: false };
+    });
+
+    setSelections(dataSources);
+  };
```

Comment:
This could be inside the other use effect directly no need to create a new function.